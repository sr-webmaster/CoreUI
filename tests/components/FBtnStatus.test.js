import createLocalVue from '../createLocalVue'
import { mount, shallowMount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FBtnStatus.vue'

const ITEMS_WITH_NUMERIC_ID = [
  {
    id: 1,
    label: 'Active',
    color: 'success'
  },
  {
    id: 2,
    label: 'Inactive',
    color: 'error'
  },
  {
    id: 3,
    label: 'Hold',
    color: 'warning'
  },
  {
    id: 4,
    label: 'Pending',
    color: 'warning'
  }
]

describe('components/Button Status', () => {
  let localVue
  describe('Visuals', () => {
    beforeEach(() => {
      localVue = createLocalVue().vue
    })
    it('snapshot default', async () => {
      const wrapper = mount(Component, { localVue })
      expect(wrapper.element).toMatchSnapshot()
    })
    it('snapshot with value', async () => {
      const wrapper = mount(Component, {
        propsData: {
          value: 'active'
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    it('snapshot with props', async () => {
      const items = [
        {
          id: 1,
          label: 'One',
          color: 'success'
        }, {
          id: 2,
          label: 'Two',
          color: 'error'
        }
      ]

      const wrapper = mount(Component, {
        localVue,
        propsData: {
          value: '1',
          items
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Computed', () => {
    describe('color', () => {
      test('returns default if activeItem.color is not viable', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            items: [
              {
                id: 'active',
                label: 'Active'
              }
            ]
          }
        })
        expect(wrapper.vm).toHaveProperty('color', 'warning')
      })
      test('returns default if activeItem is not viable', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: ''
          }
        })
        expect(wrapper.vm).toHaveProperty('color', 'warning')
      })

      test('returns selected item\'s color', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 'active'
          }
        })
        expect(wrapper.vm).toHaveProperty('color', 'success')
      })

      test('returns selected item\'s color using the items', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 1,
            items: ITEMS_WITH_NUMERIC_ID
          }
        })
        expect(wrapper.vm).toHaveProperty('color', 'success')
      })
    })
    describe('label', () => {
      test('returns default if activeItem.label is not viable', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            items: [
              {
                id: 'active'
              }
            ]
          }
        })
        expect(wrapper.vm).toHaveProperty('label', 'Pending')
      })
      test('returns default if activeItem is not viable', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: ''
          }
        })
        expect(wrapper.vm).toHaveProperty('label', 'Pending')
      })

      test('returns selected item', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 'active'
          }
        })
        expect(wrapper.vm).toHaveProperty('label', 'Active')
      })

      test('returns selected item\'s label using the items', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 1,
            items: ITEMS_WITH_NUMERIC_ID
          }
        })
        expect(wrapper.vm).toHaveProperty('label', 'Active')
      })
    })
    describe('activeItem', () => {
      test('returns undefined if value is not viable', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: null
          }
        })
        expect(wrapper.vm).toHaveProperty('activeItem')
        expect(wrapper.vm.activeItem).toBeUndefined()
      })
      test('does not throw error if value is not string', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 1
          }
        })
        expect(() => {
          expect(wrapper.vm.activeItem).toBeUndefined()
        }).not.toThrow(Error)
      })
      test('returns selected activeItem using numeric value', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue().vue,
          propsData: {
            value: 1,
            items: ITEMS_WITH_NUMERIC_ID
          }
        })
        expect(wrapper.vm.activeItem).toHaveProperty('id', 1)
      })
    })
  })
})
