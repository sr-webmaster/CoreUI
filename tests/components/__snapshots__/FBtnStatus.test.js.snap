// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button Status Visuals snapshot default 1`] = `
<div
  class="v-menu"
>
  <button
    class="v-btn theme--dark success success-text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
  Active
 
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        arrow_drop_down
      </i>
    </div>
  </button>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      class="v-list theme--light"
      role="list"
    >
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Active
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Inactive
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Hold
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Pending
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/Button Status Visuals snapshot with props 1`] = `
<div
  class="v-menu"
>
  <button
    class="v-btn theme--dark success success-text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
  One
 
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        arrow_drop_down
      </i>
    </div>
  </button>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      class="v-list theme--light"
      role="list"
    >
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            One
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Two
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/Button Status Visuals snapshot with value 1`] = `
<div
  class="v-menu"
>
  <button
    class="v-btn theme--dark success success-text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
  Active
 
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        arrow_drop_down
      </i>
    </div>
  </button>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      class="v-list theme--light"
      role="list"
    >
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Active
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Inactive
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Hold
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Pending
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
